src/prob/ne.jl

```diff
@@ -3,139 +3,155 @@ export run_ne
     # constraint_on_off_compressor_flow_expansion to disallow flow
     # if the compressor is not built
-    for i in GasModels.ids(gm,:compressor)
-        GasModels.constraint_compressor_flow_ne(gm, i)
+    for i in _GM.ids(gm, :compressor)
+        _GM.constraint_compressor_ratios(gm, i)
+        _GM.constraint_compressor_mass_flow(gm, i)
     end
 
-    for i in GasModels.ids(gm, :ne_compressor)
-        GasModels.constraint_new_compressor_flow_ne(gm, i)
+    for i in _GM.ids(gm, :ne_compressor)
+        _GM.constraint_compressor_ratios_ne(gm, i)
+        _GM.constraint_compressor_ne(gm, i)
+        _GM.constraint_compressor_mass_flow_ne(gm, i)
     end
 
-    for i in GasModels.ids(gm, :valve)
-        GasModels.constraint_valve_flow(gm, i)
+    for i in _GM.ids(gm, :valve)
+        _GM.constraint_on_off_valve_mass_flow(gm, i)
+        _GM.constraint_on_off_valve_pressure(gm, i)
     end
 
-    for i in GasModels.ids(gm, :control_valve)
-        GasModels.constraint_control_valve_flow(gm, i)
+    for i in _GM.ids(gm, :regulator)
+        _GM.constraint_on_off_regulator_mass_flow(gm, i)
+        _GM.constraint_on_off_regulator_pressure(gm, i)
     end
-
-
 end
 
 # construct the gas flow feasbility problem with demand being the cost model
-function post_ne(pm::GenericPowerModel, gm::GenericGasModel; kwargs...)
+function post_ne(pm::_PM.AbstractPowerModel, gm::_GM.AbstractGasModel; kwargs...)
     kwargs = Dict(kwargs)
-    gas_ne_weight     = haskey(kwargs, :gas_ne_weight)     ? kwargs[:gas_ne_weight] : 1.0
-    power_ne_weight   = haskey(kwargs, :power_ne_weight)   ? kwargs[:power_ne_weight] : 1.0
+    gweight = haskey(kwargs, :gas_ne_weight) ? kwargs[:gas_ne_weight] : 1.0
+    pweight = haskey(kwargs, :power_ne_weight) ? kwargs[:power_ne_weight] : 1.0
     obj_normalization = haskey(kwargs, :obj_normalization) ? kwargs[:obj_normalization] : 1.0
 
-    ## Power only related variables and constraints
+    # Power-only-related variables and constraints.
     post_tnep(pm)
 
-    #### Gas only related variables and constraints
+    # Gas-only-related variables and constraints.
     post_nels(gm)
 
-    ## Gas-Grid related parts of the problem formulation
-    for i in GasModels.ids(gm, :consumer)
+    # Gas-Power related parts of the problem formulation.
+    for i in _GM.ids(gm, :delivery)
        constraint_heat_rate_curve(pm, gm, i)
     end
 
-    ### Object function minimizes demand and pressure cost
-    objective_min_ne_cost(pm, gm; gas_ne_weight = gas_ne_weight, power_ne_weight = power_ne_weight, normalization =  obj_normalization)
+    # Objective function minimizes demand and pressure cost.
+    objective_min_ne_cost(pm, gm; gas_ne_weight=gweight,
+        power_ne_weight=pweight, normalization=obj_normalization)
 end
 
-function get_ne_solution(pm::GenericPowerModel, gm::GenericGasModel)
+function get_ne_solution(pm::_PM.AbstractPowerModel, gm::_GM.AbstractGasModel; kwargs...)
```

Comment:
Is this code still active?  Should have been made obsolete by the new solution generaiton method.